src/component/loader/DropZone.tsx

```diff
@@ -62,7 +62,7 @@ function DropZone(props) {
       const fileCollection = await fileCollectionFromFileList(files);
 
       const { nmrLoaders: filter } = preferences.current;
-      const data = await readDropFiles(fileCollection, { filter });
+      const { data } = await readDropFiles(fileCollection, { filter });
```

Comment:
I guess you should at least check for the append flag (true / false) to know if the full data state should be replaced.

Reply:
@lpatiny, I did ask about it but @hamed-musallam will make it directly in the PR